### PR TITLE
multi loan basic functionality

### PR DESCRIPTION
**Pull request by cherimarie**

### CODE REVIEW

**Comment by cherimarie** [comment]
Needs tests yet, and perhaps some thought to the 'available_to_loan' scope of Book. I think our scope would be better served with some logic, rather than trusting that the 'available' attribute is always updated correctly.

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -69,6 +69,7 @@ def clear_list
 
   def show_list
     @books = Book.where(selected: true)
+    @multi_loan_available = is_librarian? && (@books - Book.available_to_loan).empty? && (@books.length < 6)
```
Comment (cherimarie) [inline app/controllers/books_controller.rb:72]: Ensures all books in list are available to loan, and that there are less than 6. We should reference the MAX_LOANS constant from user.rb here. Is that accessible from here?

File: app/controllers/loans_controller.rb [hunk 2]
```diff
@@ -12,31 +12,48 @@ def new
...
-        redirect_to book_path(@book.id) and return 
-      end 
+        redirect_to book_path(@book.id) and return
+      end
       @loan = Loan.new(book: @book)
-    end 
+    end
+  end
+
+  def loan_multi
```
Comment (cherimarie) [inline app/controllers/loans_controller.rb]: This only loans books from the list until the patron hits their limit of 5, but doesn't have a way to convey that to user.

File: app/models/user.rb [hunk 3]
```diff
@@ -41,6 +41,10 @@ def pref_name
     preferred_first_name || name.split.first
   end
 
+  def self.able_to_borrow
```
Comment (cherimarie) [inline app/models/user.rb:54]: Alternative to doing a scope.

File: app/views/books/show_list.html.erb [hunk 4]
```diff
@@ -29,4 +29,14 @@
     </tr>
     <% end %>
   </tbody>
-</table>
\ No newline at end of file
+</table>
+
+<% if @multi_loan_available %>
+  <%= form_tag(loan_multi_path, method: "post") do %>
```
Comment (cherimarie) [inline app/views/books/show_list.html.erb]: This form needs a headline or, preferably, to be invisible until a "new loan" button is clicked.

File: app/views/books/_form.html.erb [hunk 5]
```diff
@@ -19,7 +19,7 @@
     </fieldset>
     <fieldset>
         <%= f.label :location %>
-        <%= f.text_field :location %>
+        <%= f.text_field :location, disabled: disabled %>
```
Comment (patrickdent) [inline app/views/books/_form.html.erb:22]: why is this disabled?

File: app/controllers/loans_controller.rb [hunk 6]
```diff
@@ -12,31 +12,48 @@ def new
...
     elsif params[:book_id]
       @book = Book.find_by(id: params[:book_id])
-      unless @book.available 
+      unless @book.available
         flash[:error] = "Book is Not Available"
-        redirect_to book_path(@book.id) and return 
-      end 
+        redirect_to book_path(@book.id) and return
+      end
       @loan = Loan.new(book: @book)
```
Comment (patrickdent) [inline app/controllers/loans_controller.rb:24]: Maybe we can clear the book's selected status so when they go back to selected books they can see what's left.

@book.selected = false

File: app/views/books/_form.html.erb (hunk 5 above)
Comment (cherimarie) [inline app/views/books/_form.html.erb:22]: All the fields that librarians can't edit are disabled. The value here is a boolean variable that's being set on lines 1 & 2, based on role of user.

Comment (patrickdent) [inline app/views/books/_form.html.erb:22]: Oh yea! I remember now!

**Comment by patrickdent** [comment]
I think this is ready to go, pending a little review from you!

-added tests for as many cases I could think of
-used MAX_LOANS instead of #s
-check to ensure that no weirdness - like lending zero books or nil user antics

**Comment by cherimarie** [comment]
@forplayplay - I fixed the 'good_to_borrow?' method so it was actually returning true and false, and now there's two test failures, but I just don't see where they're coming from. Ideas?

**Pull request merged**